common/djangoapps/embargo/api.py

```diff
@@ -0,0 +1,266 @@
+
+log = logging.getLogger(__name__)
+
+# Reasons a user might be blocked.
+# These are used to generate info messages in the logs.
+REASONS = {
+    "ip_blacklist": u"Restricting IP address {ip_addr} {from_course} because IP is blacklisted.",
+    "ip_country": u"Restricting IP address {ip_addr} {from_course} because IP is from country {ip_country}.",
+    "profile_country": (
+        u"Restricting user {user_id} {from_course} because "
+        u"the user set the profile country to {profile_country}."
+    )
+}
+
+
+def _from_course_msg(course_id, course_is_embargoed):
+    """
+    Format a message indicating whether the user was blocked from a specific course.
+    This can be used in info messages, but should not be used in user-facing messages.
+
+    Args:
+        course_id (unicode): The ID of the course being accessed.
+        course_is_embarged (boolean): Whether the course being accessed is embargoed.
+
+    Returns:
+        unicode
+
+    """
+    return (
+        u"from course {course_id}".format(course_id=course_id)
+        if course_is_embargoed
+        else u""
+    )
+
+
+def _log_embargo_reason(check_func, course_id, course_is_embargoed):
+    """
+    Decorator for embargo check functions that will:
+        * execute the check function
+        * check whether the user is blocked by an embargo, and if so, log the reason
+        * return a boolean indicating whether the user was blocked.
+
+    Args:
+        check_func (partial): A function that should return unicode reason if the user
+            was blocked, otherwise should return None.  This function will be passed
+            `course_id` and `course_is_embarged` kwargs so it can format a detailed
+            reason message.
+
+        course_id (unicode): The ID of the course the user is trying to access.
+
+        course_is_embargoed (boolean): Whether the course the user is trying
+            to access is under an embargo.
+
+    Returns:
+        boolean: True iff the user was blocked by an embargo
+
+    """
+
+    def _inner():
+        # Perform the check and retrieve the reason string.
+        # The reason will be `None` if the user passes the check and can access the course.
+        # We pass in the course ID and whether the course is embargoed
+        # so that the check function can fill in the "reason" message with more specific details.
+        reason = check_func(
+            course_id=course_id,
+            course_is_embargoed=course_is_embargoed
+        )
+
+        # If the reason was `None`, indicate that the user was not blocked.
+        if reason is None:
+            return False
+
+        # Otherwise, log the reason the user was blocked
+        # and return True.
+        else:
+            msg = u"Embargo: {reason}".format(reason=reason)
+            log.info(msg)
+            return True
+
+    return _inner
+
+
+def _embargo_redirect_response():
```

Comment:
We're going to need to replace this with a redirect to new "message" views (in another PR I recently merged).  For now, please use the same `_embargo_redirect_response` implementation as the old implementation (in middleware.py).